lib/livebook/apps/manager.ex

```diff
@@ -0,0 +1,245 @@
+defmodule Livebook.Apps.Manager do
+  # Orchestrates permanent app deployments across the cluster.
+  #
+  # Only a single instance of the manager runs in the cluster. This is
+  # ensured by registering it in :global. Each node also runs a single
+  # instance of `Livebook.Apps.ManagerWatcher`, which starts a new
+  # manager whenever needed.
+
+  use GenServer, restart: :temporary
+
+  require Logger
+
+  alias Livebook.Apps
+
+  @name __MODULE__
+
+  config = Application.compile_env(:livebook, __MODULE__)
+
+  @handle_app_close_debounce_ms 100
+  @retry_backoff_base_ms Keyword.fetch!(config, :retry_backoff_base_ms)
+
+  @doc """
+  Starts a new manager process.
+  """
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, {}, name: {:global, @name})
+  end
+
+  @doc """
+  Checks if the manager is running on the current node.
+  """
+  @spec local?() :: boolean()
+  def local?() do
+    case :global.whereis_name(@name) do
+      :undefined -> false
+      pid -> node(pid) == node()
+    end
+  end
+
+  @doc """
+  Asynchronously asks the manager to reflect the currently configured
+  permanent apps.
+
+  Any permanent apps that are not running will be deployed. Apps that
+  should no longer be running will be closed.
+  """
+  @spec sync_permanent_apps() :: :ok
+  def sync_permanent_apps() do
+    GenServer.cast({:global, @name}, :sync_permanent_apps)
+  end
+
+  @impl true
+  def init({}) do
+    Apps.subscribe()
+
+    state = %{deployments: %{}, handle_app_close_timer_ref: nil}
+    state = sync_apps(state)
```

Comment:
Should we sync apps here? Or should we send a message to this process after setting up the persistent term on the application start callback?